frontend/views/components/GroupSettings/InviteLinks.vue

```diff
@@ -172,9 +126,18 @@ export default {
     }
   },
   computed: {
-    activeList () {
-      return this.ephemeral.selectbox.selectedOption === 'Active' ? this.ephemeral.dummyInviteList.filter(item => !item.state.isExpired)
-        : this.ephemeral.selectbox.selectedOption === 'All' && this.ephemeral.dummyInviteList
+    ...mapGetters(['currentGroupState']),
+    ...mapState(['currentGroupId']),
+    invitesToShow () {
+      const { invites } = this.currentGroupState
+      const invitesList = Object.entries(invites).map(([inviteSecret, invite]) => ({
+        inviteSecret,
+        inviteLink: buildInvitationUrl(this.currentGroupId, inviteSecret),
+        ...invite
+      }))
+
+      return this.ephemeral.selectbox.selectedOption === 'Active' ? invitesList.filter(invite => invite.status === 'valid')
+        : this.ephemeral.selectbox.selectedOption === 'All' && invitesList
```

Comment:
To avoid this complex ternary, you can use a different approach with objects. Besides being faster, I find it easier to understand as well.

```js
// each key is a function, so the code inside is only executed if called.
const options = {
 Active: () => invitesList.filter(invite => invite.status === 'valid'),
 All: () => invitesList
}

return option[this.ephemeral.selectbox.selectedOption]()
```

Reply:
I agree it looks much neater that way. fixed! 👍